container/appmaster/runit/service/flink/run

```diff
@@ -73,13 +73,37 @@ function add_kerberos_configurations() {
   fi
 }
 
+function add_ha_configurations() {
+  if [[ "${FLINK_HA_ENABLED}" == true ]]; then
+    export FLINK_JAVA_OPTS="$FLINK_JAVA_OPTS -Dhigh-availability=ZOOKEEPER"
```

Comment:
Have a look at the `add_if_non_empty` function, it is basically a wrapper for appending to `FLINK_JAVA_OPTS`.   For the sake of uniformity, please use it to set the `high-availability` flag.

I have a minor concern about the sheer number of settings being configured as dynamic properties, as opposed to being configured in `flink-conf.yaml`.  Perhaps some of these HA settings can be set statically.